01-node-tutorial/answers/prompter.js

```diff
@@ -21,34 +22,72 @@ const getBody = (req, callback) => {
 };
 
 // here, you could declare one or more variables to store what comes back from the form.
-let item = "Enter something below.";
-
+let heading = "Life journal";
+let title = "New title";
+let thoughts = "New thoughts";
+let color = "white";
 // here, you can change the form below to modify the input fields and what is displayed.
 // This is just ordinary html with string interpolation.
+
 const form = () => {
   return `
-  <body>
-  <p>${item}</p>
+  <body style="background-color:${color}">
   <form method="POST">
-  <input name="item"></input>
-  <button type="submit">Submit</button>
+
+  <div style="display: flex; flex-direction: column; width:20rem;">
+    <h1>${heading}</h1>
+
+    <h3>Title:</h3>
+    <input name="title"></input>
+    <h3>Your thoughts:</h3>
+    <textarea name="thoughts"></textarea>
+    <label for="color" style="margin-top:1rem">Choose a background color</label>
+    <select name="color" id="color" >
+    <option value="color">All Colors</option>
+    <option value="white">White</option>
+    <option value="red">Red</option>
+    <option value="green">Green</option>
+    <option value="blue">Blue</option>
+    <option value="yellow">Yellow</option>
+    </select>
+    <button type="submit" style="width:5rem; margin-top:1rem">Submit</button>
+   
+    <p>${title}</p>
+    <p>${thoughts} </p>
+  </div>
   </form>
   </body>
   `;
 };
-
 const server = http.createServer((req, res) => {
   console.log("req.method is ", req.method);
   console.log("req.url is ", req.url);
   if (req.method === "POST") {
     getBody(req, (body) => {
       console.log("The body of the post is ", body);
       // here, you can add your own logic
-      if (body["item"]) {
-        item = body["item"];
-      } else {
-        item = "Nothing was entered.";
+
+      if (body.color) {
+        color = body.color;
       }
+
+      const records = {
+        title: body["title"] || "No title provided",
+        thoughts: body["thoughts"] || "No thoughts provided",
+        date: new Date().toISOString(),
+      };
+      fs.writeFile(
```

Comment:
### Informational
[Good Stuff 👍 ]
This is a great improvement, a little logging for your application is always a good deal.

You also correctly used async here which means the end-user doesn't have to wait for you to write to file to get their response 🎉 . We sometimes call this "fire and forget" or "out of band" or just async logging in this instance. The idea being while logging is important it doesn't need to stand in the way of the end user experience (unless of course there's a security reason we need to know 100% that the logging happened).

Reply:
Thank you